### ISSUE TITLE
System.Drawing.Imaging.Metafile(Stream) throws wrong Exception when Stream is Null on netcoreapp

### ISSUE THREAD

**Issue by KostaVlev** [dotnet/corefx#22741]
**Test Null Stream:**

```c#
[ConditionalFact(nameof(PlatformDetection) + "." + nameof(PlatformDetection.IsNotWindowsNanoServer))]
public void Ctor_NullStream_ThrowsArgumentException()
{
    AssertExtensions.Throws<ArgumentException>(null, () => new Metafile((Stream)null));
}
```

Expected: `System.ArgumentException`

Actual:
netfx `System.ArgumentException`
netcoreapp `System.ArgumentNullException`

**Edit**
Forgot to reference related issue  (#20711)

**Comment by JonHanna** [comment]
Throwing a more derived exception is an improvement the [breaking change rules](https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/breaking-change-rules.md#exceptions) allow.

**Comment by mellinoe** [comment]
https://github.com/dotnet/corefx/commit/cdfab0fb69392950ab7cc3926671c846401ddfd5 did some cleanup and changed a few `ArgumentException`s to `ArgumentNullException` (@hughbe). Like @JonHanna said, it's a compatible change, although honestly I would prefer to just keep the behavior the same in this case, given that System.Drawing.Common is around for legacy compat. In this case, I'm not going to revert the change -- it's too minor and insignificant.

**Comment by mellinoe** [comment]
Actually -- I'll leave this open until I change the test.

**Comment by KostaVlev** [comment]
Fixed by #22941
